src/parameterizations/vertical/MOM_diabatic_driver.F90

```diff
@@ -1060,6 +1060,13 @@ subroutine diabatic(u, v, h, tv, fluxes, visc, ADp, CDp, dt, G, GV, CS)
     enddo ; enddo ; enddo
   endif
 
+  ! Undo the effects of applyBoundaryFluxesInOut for passive tracers
+  if (CS%useALEalgorithm) then
+    k = 1
+    do j=js,je ; do i=is,ie
+      hold(i,j,k) = hold(i,j,k) - ea(i,j,k)
```

Comment:
True that would lead to some very not good things. Maybe instead t he fix should be to set 'ea(i,j,k=1)' to 'dThickness'  in subroutine:applyBoundaryFluxesInOut (which does guard against negative thickness values). However, this also raises a question for me as how to cast a change in layer thickness in any layer other than the top layer from boundary fluxes in terms of 'ea' and 'eb' (which seems like it could happen if MINIMUM_FORCING_DEPTH is larger than the first layer). Thoughts?